src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/finance/PrisonBalanceReconciliationService.kt

```diff
@@ -53,9 +56,11 @@ class PrisonBalanceReconciliationService(
       }
   }
 
-  suspend fun generateReconciliationReport(prisonIds: List<String>): List<MismatchPrisonBalance> = withContext(Dispatchers.Unconfined) {
-    prisonIds.map { async { checkPrisonBalanceMatch(it) } }
-  }.awaitAll().filterNotNull()
+  suspend fun generateReconciliationReport(prisonIds: List<String>): List<MismatchPrisonBalance> = prisonIds.chunked(pageSize.toInt()).flatMap { pagedPrisonIds ->
```

Comment:
adding paging to avoid bombarding nomis-prisoner-api causing `Could not open JPA EntityManager for transaction`